### PR TITLE
fix: Cart not opening as a menu item

### LINKED ISSUE #8558: Cart not opening as a menu item
- it opens under the burger menu 
image

not really ideal 
image

try anywhere where navbar is burger menu

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [x] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Moved  viewport logic to `useViewport.ts`
- [x] Closes #8558

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

🐞 Bug

https://github.com/kodadot/nft-gallery/assets/44554284/3d1e77e4-0ed5-4f2e-a8bb-385a2c589bec

❤️‍🩹 Fixed

https://github.com/kodadot/nft-gallery/assets/44554284/fc92e74f-633f-4765-80c4-466c8362b3d2

### CODE REVIEW

File: composables/useViewport.ts [hunk 1]
```diff
@@ -1,7 +1,14 @@
 import { useWindowSize } from '@vueuse/core'
 
 export default function () {
-  const isMobile = computed(() => useWindowSize().width.value <= 768)
+  const { width } = useWindowSize()
```
Comment (preschian) [inline composables/useViewport.ts:4]: I already added this module https://www.npmjs.com/package/@nuxtjs/device. I think it's better to use that instead

**Review by preschian — CHANGES_REQUESTED** [review]

**Review by roiLeo — COMMENTED** [review]
On the other hand I would like that we use tailwind responsive helpers instead when possible (#8767)

File: composables/useViewport.ts (hunk 1 above)
Comment (hassnian) [reply composables/useViewport.ts:4]: but this module checks the device type

using `isMobile` for example on a desktop with a window like this 
CleanShot 2024-01-25 at 13 39 21@2x

returns `false` this is not what we want in this case

Comment (preschian) [reply composables/useViewport.ts:4]: i dont think common user have this size for a desktop

**Comment by hassnian** [comment]
> On the other hand I would like that we use tailwind responsive helpers instead when possible (#8767)

@roiLeo  done pls check 

> i dont think common user have this size for a desktop

@preschian  yep. I agree I just wanted to point out, that on a desktop if the user resizes the window since we are no longer using the screen size it wouldn't behave as it did before 

CleanShot 2024-01-26 at 11 40 21@2x
   
as I see it the before and after are not the same, but I might be missing something.

thanks

**Review by roiLeo — CHANGES_REQUESTED** [review]
Something wrong with navbat on mobile compared to canary

Capture d’écran 2024-01-26 à 8 09 49 AM

**Comment by hassnian** [comment]
> Something wrong with navbat on mobile compared to canary
> 
> Capture d’écran 2024-01-26 à 8 09 49 AM

> Something wrong with navbat on mobile compared to canary 

> but this module checks the device type
> 
> using `isMobile` for example on a desktop with a window like this CleanShot 2024-01-25 at 13 39 21@2x
> 
> returns `false` this is not what we want in this case

this is what I meant.  

with the device type set to mobile one would work fine - issue preview vs canary

https://github.com/kodadot/nft-gallery/assets/44554284/f017199e-7824-419a-8e21-8074d7bd678b

cc @preschian

**Comment by roiLeo** [comment]
> this is what I meant.
> 
> with the device type set to mobile one would work fine - issue preview vs canary

sorry about that, don't you think we need to go inside each components to update js -> css with visibility helpers?

File: assets/styles/global.scss [hunk 2]
```diff
@@ -288,3 +288,11 @@ hr {
 .min-width-0 {
   min-width: 0;
 }
+
+.is-desktop-hidden {
+  @apply lg:hidden #{!important};
```
Comment (preschian) [inline assets/styles/global.scss:293]: Please avoid adding more classes on global.scss. You can use `lg:!hidden` on tailwind directly

**Review by preschian — COMMENTED** [review]
sorry guys, I dont get it the issue. is this about the logo?

File: components/Navbar.vue [hunk 3]
```diff
@@ -240,7 +234,7 @@ const account = computed(() => identityStore.getAuthAddress)
 const isCreateVisible = computed(() => createVisible(urlPrefix.value))
 
 const logoSrc = computed(() => {
-  const variant = isMobile.value ? 'Koda' : 'Koda_Beta'
+  const variant = isTouch ? 'Koda' : 'Koda_Beta'
```
Comment (preschian) [inline components/Navbar.vue:237]: If you want to address #8767, this line isn't needed. You can wrap HTML something like this

```vue
<nuxt-link to="/" class="navbar-item logo nuxt-link-active">
  <!-- logo on mobile -->
  <img
    src="/Koda.svg"
    alt="First NFT market explorer on Kusama and Polkadot"
    class="lg:hidden" />
	<!-- logo on desktop -->
  <img
    src="/Koda_Beta.svg"
    alt="First NFT market explorer on Kusama and Polkadot"
    class="hidden lg:block" />
</nuxt-link>
```

Comment (preschian) [reply components/Navbar.vue:237]: you just need to wrap it for dark mode

File: assets/styles/global.scss (hunk 2 above)
Comment (hassnian) [reply assets/styles/global.scss:293]: duly noted

**Review by preschian — APPROVED** [review]
lgtm. It can be a follow-up issue for the CSS helper (`isMobile`). Related:
- https://github.com/kodadot/nft-gallery/issues/8767
- https://github.com/kodadot/nft-gallery/issues/8489

**Review by roiLeo — APPROVED** [review]

**Comment by prury** [comment]
some link texts broke tho: 
image

and problem still persists here for me

**Comment by hassnian** [comment]
> some link texts broke tho: image
> 
> and problem still persists here for me

pls check https://github.com/kodadot/nft-gallery/pull/9131#issuecomment-1911599979 https://github.com/kodadot/nft-gallery/pull/9131#discussion_r1466028247

this is to be expected now 

you have to test it with device set to `mobile` and not  a `desktop` window with a small screen

https://github.com/kodadot/nft-gallery/assets/44554284/16f50e48-8356-4343-9123-9ce5a8ecf1c2

**Comment by prury** [comment]
> > some link texts broke tho: image
> > and problem still persists here for me
> 
> @prury pls check [#9131 (comment)](https://github.com/kodadot/nft-gallery/pull/9131#discussion_r1466028247) [#9131 (comment)](https://github.com/kodadot/nft-gallery/pull/9131#issuecomment-1911599979)
> 
> this is to be expected now
> 
> you have to test it with device set to `mobile` and not a `desktop` window with a small screen
> CleanShot.2024-01-29.at.18.19.00.mp4

oh ok ok, it works then, brain still syncing up

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

testing new token for bot hope it works

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 6.97 USD/DOT ~ 4.304 $DOT
🧗 [16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD](https://polkadot.subscan.io/account/16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)
🔗 [0x1efdd5f3f4b37b08dfdbffc59d163019708662541cf1fa423c3c6fd2f0125165](https://polkadot.subscan.io/extrinsic/0x1efdd5f3f4b37b08dfdbffc59d163019708662541cf1fa423c3c6fd2f0125165)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**